### PR TITLE
Restrict same usergroup create multiple times in a project

### PR DESCRIPTION
**Pull request by k9845**
Fixes:
- #740 New Project Flow: Handle error while adding same user-group twice in a project

### CODE REVIEW

File: apps/project/serializers.py [hunk 1]
```diff
@@ -459,10 +459,20 @@ class Meta:
...
     def validate(self, data):
         data['project_id'] = int(self.context['view'].kwargs['project_id'])
+        usergroup = data.get('usergroup')
+        usergroup = self.validate_project_usergroup(data['project_id'], usergroup)
         return data
 
+    def create(self, validated_data):
+        instance, _ = ProjectUserGroupMembership.objects.get_or_create(**validated_data)
+        return instance
+
```
Comment (thenav56) [inline apps/project/serializers.py]: This should be enough right?. Also for the patch method user group can be None.

```suggestion
    def validate(self, data):
        data['project_id'] = int(self.context['view'].kwargs['project_id'])
        usergroup = data.get('usergroup')
         if usergroup and ProjectUserGroupMembership.objects.filter(project=data['project_id'], usergroup=usergroup).exists():
            raise serializers.ValidationError({'usergroup': 'Usergroup already exist in the project'})
        return data
```

File: apps/project/tests/test_apis.py [hunk 2]
```diff
@@ -538,6 +538,31 @@ def test_update_project_remove_ug(self):
...
+
+        # now try to create same usergroup from api level
+        data = {
+            'usergroup': self.ug1.id,
+            'role': self.normal_role.id
+        }
+        url = f'/api/v1/projects/{project.id}/project-usergroups/'
+        self.authenticate()
+        response = self.client.post(url, data)
+        self.assert_400(response)
```
Comment (thenav56) [inline apps/project/tests/test_apis.py:565]: Let's add a retry after removing the membership we added manually.

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/project/tests/test_apis.py (hunk 2 above)
Comment (k9845) [reply apps/project/tests/test_apis.py:565]: Done!

File: apps/project/serializers.py (hunk 1 above)
Comment (k9845) [reply apps/project/serializers.py]: Done!!

**Review by thenav56 — APPROVED** [review]

**Review by annshress — APPROVED** [review]

**Pull request merged**